clang/lib/Interpreter/IncrementalExecutor.cpp

```diff
@@ -73,7 +74,15 @@ llvm::Error IncrementalExecutor::addModule(PartialTranslationUnit &PTU) {
       Jit->getMainJITDylib().createResourceTracker();
   ResourceTrackers[&PTU] = RT;
 
-  return Jit->addIRModule(RT, {std::move(PTU.TheModule), TSCtx});
```

Comment:
Even if we are attaching a resource tracker (`RT`) the Jit decides to destroy the `ThreadSafeModule`.